storagenode/piecestore/endpoint.go

```diff
@@ -522,8 +523,9 @@ func (endpoint *Endpoint) Download(stream pb.Piecestore_DownloadServer) (err err
 			// TODO: add timeout here
 			message, err = stream.Recv()
 			if err != nil {
-				// err is io.EOF when uplink closed the connection, no need to return error
-				return ErrProtocol.Wrap(ignoreEOF(err))
+				// err is io.EOF or canceled when uplink closed the connection, no need to return error
+				err = errs2.IgnoreCanceled(ignoreEOF(err))
```

Comment:
Do not ignore this error. When the uplink cancels the download we need to log it for debugging purposes.

Reply:
Now if canceled, there is a debug log